Add Terraform Configurations for Minikube and GKE

jfaltermeier opened the pull request:
We want to use Terraform for providing ready-to-use Getting Started setups to new users. 
User may simply use `terraform apply` to create the required infrastructure and get a Theia Cloud sample installation. 
This should also help reproducibility for issues, as we ask for the configuration that was used. 

This PR adds modules for create a Minikube or GKE cluster, installing Theia Cloud + Dependencies, and setting up a Keycloak Realm. 

Those modules are combined to offer a Minikube and a GKE Getting Started configuration. 
We may add more configurations in the future.

lucas-koehler commented on terraform/terraform.md:
```diff
@@ -0,0 +1,117 @@
+
+The `modules` directory contains our reuseable terraform modules for creating clusters, installing dependencies via helm, and configuring keycloak. The modules will be used by the actual terraform configurations available in the `configurations` directory.
+
+If you can't use terraform, the `./modules/helm/main.tf` contains the information which helm charts are installed from which helm repository and you may extract the passed values. For and initial Keycloak realm configuration, you may check the values in `./modules/keycloak/main.tf`.
```
```suggestion
If you can't use Terraform, the `./modules/helm/main.tf` contains the information which helm charts are installed from which helm repository and you may extract the passed values. For an initial Keycloak realm configuration, you may check the values in `./modules/keycloak/main.tf`.
```

lucas-koehler commented on terraform/terraform.md:
```diff
@@ -0,0 +1,117 @@
+
+## Theia Cloud Getting Started
+
+Currently we have Getting Started configuration for the following Clusters/Services:
```
```suggestion
Currently, we have Getting Started configurations for the following Clusters/Services:
```

lucas-koehler reviewed (commented):
Looks pretty good to me already, thanks! I added some inline suggestions to fix minor typos/grammar.

I noticed these empty files:
1. `terraform/modules/helm/outputs.tf`
2. `terraform/modules/keycloak/outputs.tf`
Do we need them?

jfaltermeier commented:
> Looks pretty good to me already, thanks! I added some inline suggestions to fix minor typos/grammar. I tested the Minikube setup locally and it worked very well. I did not test GKE.
> 
> I noticed these empty files:
> 
> 1. `terraform/modules/helm/outputs.tf`
> 2. `terraform/modules/keycloak/outputs.tf`
>    Do we need them?

I've removed the empty files. 
Also I've fixed an issue with minikube that `eval $(minikube docker-env)` was not working. 

You may test the GKE configuration with our GKE kubernetes project and run `destroy` again after testing. The configuration should produce a valid hostname, so no manual DNS configuration should be required.

lucas-koehler reviewed (approved):
Thanks for the updates:) I also tested the GKE config now and it worked well :)
I also found out that the terraform has a formatter. What do you think of automatically formatting terraform files on save?
For me this worked in VS Code by configuring this in the workspace settings by adding file `.vscode/settings.json` with this content:
```json
{
  "[terraform]": {
    "editor.defaultFormatter": "hashicorp.terraform",
    "editor.formatOnSave": true,
  },
  "[terraform-vars]": {
    "editor.defaultFormatter": "hashicorp.terraform",
    "editor.formatOnSave": true,
  }
}
```

lucas-koehler commented:
@jfaltermeier Thanks for adding the settings. This is fine for me to be merged then :tada:
